fix: Capture correct descriptions for RSpec one-liner examples

americodls opened the pull request:
RSpec one-liner examples (e.g., `it { is_expected.not_to be_nil }`) generate descriptions after execution.

Updated formatter to capture the final description in `example_finished` instead of relying on the incomplete description at `example_started`.

americo-duarte committed:
fix: capture correct descriptions for RSpec one-liner examples

RSpec one-liner examples (e.g., `it { is_expected.not_to be_nil }`)
generate descriptions after execution. Updated formatter to capture
the final description in `example_finished` instead of relying on
the incomplete description at `example_started`.

kaylareopelle reviewed (approved):
Hi @americodls, thanks for catching this and providing a fix!

arielvalentin committed:
Merge branch 'main' into ad/fix-rspec-one-liner-descriptions

arielvalentin committed:
Merge branch 'main' into ad/fix-rspec-one-liner-descriptions

arielvalentin commented on instrumentation/rspec/lib/opentelemetry/instrumentation/rspec/formatter.rb:
```diff
@@ -68,8 +68,12 @@ def example_started(notification)
 
+            # Update name and full_description for one-liner examples where description is generated after execution
+            span.name = example.description
+            span.set_attribute('rspec.example.full_description', example.full_description.to_s)
```
💁🏼 Use `add_attributes` as opposed to setting each attribute separately because each call requires a lock when changing a span

kaylareopelle replied:
Great catch. I agree. @americodls, can you update the method call to combine the attributes on line 76 and 77 into one `add_attributes` call?

americodls replied:
Thank you for the feedback. I just fixed that.

americodls committed:
Merge remote-tracking branch 'origin' into ad/fix-rspec-one-liner-descriptions

americodls committed:
Batch span attributes to reduce lock contention

Using add_attributes instead of two separate set_attribute calls
reduces lock acquisition overhead, since each attribute update requires
locking the span.
